corehq/form_processor/submission_post.py

```diff
@@ -649,3 +658,18 @@ def notify_submission_error(instance, message, exec_info=None):
     }
     request = get_request()
     notify_exception(request, message, details=details, exec_info=exec_info)
+
+
+@quickcache(['domain', 'build_id'])
+def _get_form_name_info(domain, build_id):
+    """
+    :return: (default_language, {'http://my.xmlns': {'en': 'My Form'}})
```

Comment:
This is kind of a clumsy return value.  It would've been simpler to vary the cache by `domain, build_id, xmlns, user_lang`, but I opted to cache all the form names for each app - fewer slightly larger caches versus many tiny ones.  Open to second opinions there.